src/renderer/components/dock/dock-tabs.tsx

```diff
@@ -39,14 +99,56 @@ export const DockTabs = ({ tabs, autoFocus, selectedTab, onChangeTab }: Props) =
     }
   };
 
+
+  useEffect(() => {
+    updateStateValues();
```

Comment:
Instead of this you could also add `{fireImmediately: true}` for your reaction (to apply initial values + watch changes at the same time)